src/base/detoken.lua

```diff
@@ -56,16 +57,26 @@
 			setfenv(func, e)
 
 			-- run it and get the result
-			local result = func() or ""
+			local success, result = pcall(func)
+			if not success then
+				err    = result
+				result = nil
+			else
+				err    = nil
```

Comment:
Your call, there's merits both ways, less code is always good, but it's very explicit that `err` is `nil` with the else there which is good.